node/flatpak-node-generator.py

```diff
@@ -247,18 +248,18 @@ def __get_cache_bucket(self, cachable: bool, url: str) -> Cache.BucketRef:
 
     async def _read_parts(self,
                           url: str,
-                          size: int = DEFAULT_PART_SIZE) -> AsyncIterator[bytes]:
+                          size: int = DEFAULT_PART_SIZE, headers: dict[str,str] = None) -> AsyncIterator[bytes]:
```

Comment:
Two primary issues here:

- using the builtin types with `[]` syntax is invalid prior to Python 3.9, this should be `Dict[str, str]`
- default should be `{}`, `None` won't actually type check and also isn't entirely valid (I'm not sure if this worked due to sheer coincidence, but the defaults in urllib use `{}`, so relying on `None` working might not be guaranteed)